text/0000-action-menu.md

```diff
@@ -0,0 +1,214 @@
+
+# Detailed design
+
+The newly proposed `ActionMenu` embodies a headless component. This means it provides the functional logic of a menu without imposing any default styles. It is a composite of multiple sub-components, allowing it to adapt to and satisfy the requirements of any application. Moreover, it incorporates three paramount principles: accessibility (A11y), high performance, and ease of use.
+
+1. **Accessibility (A11y)**: Prioritizing accessibility, the `ActionMenu` is designed to ensure that all users, regardless of their abilities, can smoothly interact with it. Every component part will incorporate accessibility standards as per the Web Content Accessibility Guidelines (WCAG) to ensure the platform's inclusivity.
+
+2. **High Performance**: The `ActionMenu` is optimized for high performance. It's designed to be lightweight and responsive, ensuring fast loading times and a smooth user experience, irrespective of the application's scale.
+
+3. **Ease of Use**: While offering complex functionality, the `ActionMenu` is designed with user-friendliness in mind. Developers can easily combine its components to create a custom ActionMenu tailored to their application's needs.
+
+Here is a rundown of the `ActionMenu` structure:
+
+- **`ActionMenu`**: This component acts as a wrapper for all other components, maintaining the state of the menu. Its design provides a flexible user interface that can be adapted to suit the unique requirements of any application, while preserving optimal accessibility.
+
+- **`ActionMenu.Button`**: This component is the button that triggers the menu. Developers can define the appearance of the button, allowing for a high degree of visual customization while maintaining ease of interaction.
+
+- **`ActionMenu.Overlay`**: This component encapsulates the `ActionList` and is the actual menu displayed when the `ActionMenu.Button` is clicked. It is designed with user accessibility in mind, ensuring that the menu is easily navigable.
+
+- **`ActionList`**: This component acts as a wrapper for the menu items, allowing developers to customize the arrangement of items to suit their specific needs.
+
+- **`ActionList.Item`**: This represents a single menu item. Each item is individually customizable, ensuring a balance between user interaction and accessibility.
+
+The proposed design of the `ActionMenu` ensures that it is not only versatile and easy to use but also maintains an inclusive, accessible interface for all users while still being performant.
+
+# Drawbacks
+
+Creating an ActionMenu for a design system does come with its own set of potential drawbacks. Here are a few:
+
+**Complexity and Resource Requirements**: Designing a flexible and widely usable ActionMenu can be a complex task, demanding substantial time and resources. Considering the diverse requirements of different applications, it's challenging to develop a solution that satisfies all possible needs.
```

Comment:
Disagree here. Flexibility all depends on the design. When building the components correctly, you can make anything flexible for any use case. 😉

Reply:
This has been removed.